### PR TITLE
Feature/lp 109 ProfileViewModel

### PR DESCRIPTION
**Pull request by JoseGR98**
Profile - ViewModel
[LP-109](https://devaptivist.atlassian.net/browse/LP-109)
View Model implementation for the Profile View.
Files added/edited:
- ProfileView that contains the UI
- ProfileViewModel that handles the logic of data
- ProfileModel that contains the data class for the User Profile Data
NavGraph of Profile edited because of new parameter -  TODO

### CODE REVIEW

File: core/data/src/main/java/com/iteneum/ProfileModel.kt
Comment (irv205) [inline core/data/src/main/java/com/iteneum/ProfileModel.kt]: Rename "ProfileModel" to "Profile"

File: core/navigation/src/main/java/com/iteneum/navigation/graphs/DashBoardNavGraph.kt
Comment (irv205) [inline core/navigation/src/main/java/com/iteneum/navigation/graphs/DashBoardNavGraph.kt]: Explain in that TODO what you need to do in that specific part.

File: feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt
Comment (irv205) [inline feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt]: rename this parameter, it depends what do you want, if it's to navigate back you can rename it as "BackToDashBoard" or if it's to navigate to other view you can rename "navigateToNameView"

File: core/data/src/main/java/com/iteneum/ProfileModel.kt
Comment (JoseGR98) [reply core/data/src/main/java/com/iteneum/ProfileModel.kt]: Sure, no problem. Done!

File: core/navigation/src/main/java/com/iteneum/navigation/graphs/DashBoardNavGraph.kt
Comment (JoseGR98) [reply core/navigation/src/main/java/com/iteneum/navigation/graphs/DashBoardNavGraph.kt]: Got it! Done!

File: feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt
Comment (JoseGR98) [reply feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt]: It is for going back, by pressing the arrow component, but not implemented yet. Changed the name for the one you mentioned. Thank you!

**Review by irv205 — APPROVED** [review]

File: feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt
Comment (carlosph816) [inline feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: we can change this in the same way you did with profileModel

File: feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt
Comment (carlosph816) [inline feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt]: we don't need to give the viewmodel here, you need to implement the hight order functions for events

File: feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt
Comment (JoseGR98) [reply feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: Ok, done!

File: feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt
Comment (JoseGR98) [reply feature/profile/src/main/java/com/iteneum/profile/presentation/ProfileView.kt]: Got it, done!

**Comment by JoseGR98** [comment]
New PR made. Code review pending

File: feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt
Comment (carlosph816) [inline feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: this shouldn't be here, let's delete and put a "TODO"

File: feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt
Comment (carlosph816) [inline feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: this shouldn't be here, let's delete and put a "TODO"

Comment (JoseGR98) [reply feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: Got it. Modified!

File: feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt
Comment (JoseGR98) [reply feature/profile/src/main/java/com/iteneum/profile/data/ProfileViewModel.kt]: Got it. Modified!

**Comment by JoseGR98** [comment]
1 code reviewer approved. Waiting for one more approval from you @carlosph816 
Both approved! Only merge left! :D

**Review by carlosph816 — APPROVED** [review]

**Pull request merged**
